src/change/sdp.rs

```diff
@@ -73,7 +74,14 @@ impl<'a> SdpApi<'a> {
             return Err(RtcError::RemoteSdp("No m-lines in offer".into()));
         }
 
-        add_ice_details(self.rtc, &offer)?;
+        // When the remote offers, it indicates an ICE restart by changing its ICE credentials
+        let ice_restart = self
+            .rtc
+            .ice
+            .remote_credentials_changed(offer.ice_creds().as_ref());
+        // we always accept the ICE restart by providing new credentials ourselves
+        let maybe_new_ice_creds = ice_restart.then(IceCreds::new);
+        add_ice_details(self.rtc, &offer, ice_restart, maybe_new_ice_creds.as_ref())?;
```

Comment:
We should keep `add_ice_details` as is, and move the logic for detecting ice_restart to inside `add_ice_details`.

Reply:
I was trying to do that initially but the logic is different for answers and offers, so I kept as much common as possible. We could pass down the remote ice credentials too making the signature:

```rust
fn add_ice_details(
    rtc: &mut Rtc,
    sdp: &Sdp,
    remote_ice_creds: Option<&IceCreds>,
    new_local_ice_creds: Option<&IceCreds>,
);
```

for a slight improvement tough

Reply:
In what way is it different?

Reply:
Addressed in e794138